postgres/datadog_checks/postgres/statements.py

```diff
@@ -311,20 +304,23 @@ def _explain_new_pg_stat_activity(self, samples, seen_statements, seen_statement
         return events
 
     def _collect_execution_plans(self, instance_tags):
+        if not self._activity_sample_rate_limiter:
+            self._activity_sample_rate_limiter = ConstantRateLimiter(self.config.collect_exec_plans_rate_limit)
+
         start_time = time.time()
         # avoid reprocessing the exact same statement
         seen_statements = set()
         # keep only one sample per unique (query, plan)
         seen_statement_plan_sigs = set()
-        while time.time() - start_time < self.config.collect_exec_plan_time_limit:
-            if len(seen_statement_plan_sigs) > self.config.collect_exec_plan_event_limit:
+        while time.time() - start_time < self.config.collect_exec_plans_time_limit:
+            if len(seen_statement_plan_sigs) > self.config.collect_exec_plans_event_limit:
                 break
             samples = self._get_new_pg_stat_activity(instance_tags=instance_tags)
             events = self._explain_new_pg_stat_activity(samples, seen_statements, seen_statement_plan_sigs,
                                                         instance_tags)
             if events:
                 submit_exec_plan_events(events, instance_tags, "postgres")
-            time.sleep(self.config.collect_exec_plan_sample_sleep)
+            self._activity_sample_rate_limiter.sleep()
```

Comment:
As noted above, I think this should block at the head of the while loop, not the end.

Reply:
good point, moved to the top